components/encryption/src/master_key/kms.rs

```diff
@@ -16,30 +15,70 @@ use tokio::runtime::{Builder, Runtime};
 
 use super::{metadata::MetadataKey, Backend, MemAesGcmBackend};
 use crate::config::KmsConfig;
-use crate::crypter::Iv;
+use crate::crypter::{Encrypted, EncryptedRef, Iv};
 use crate::{Error, Result};
 use rusoto_util::new_client;
 
 const AWS_KMS_DATA_KEY_SPEC: &str = "AES_256";
 const AWS_KMS_VENDOR_NAME: &[u8] = b"AWS";
 
-struct AwsKms {
+#[derive(PartialEq, Debug)]
+pub struct KeyId(String);
+
+// KeyID is a newtype to mark a String as an ID of a key
+// This ID exists in a foreign system such as AWS
+// The key id must be non-empty
+impl KeyId {
+    fn new(id: String) -> Result<KeyId> {
+        if id.is_empty() {
+            Err(box_err!("KMS key id can not be empty"))
+        } else {
+            Ok(KeyId(id))
+        }
+    }
+}
+
+pub struct AwsKms {
     client: KmsClient,
-    current_key_id: String,
+    current_key_id: KeyId,
     runtime: Runtime,
-    // The current implementation (rosoto 0.43.0 + hyper 0.13.3) is not `Send`
-    // in practical. See more https://github.com/tikv/tikv/issues/7236.
-    // FIXME: remove it.
-    _not_send: PhantomData<*const ()>,
+    region: String,
+    endpoint: String,
+}
+
+struct KmsClientDebug {
+    region: String,
+    endpoint: String,
+}
+
+impl std::fmt::Debug for KmsClientDebug {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("KmsClient")
+            .field("region", &self.region)
+            .field("endpoint", &self.endpoint)
+            .finish()
+    }
+}
+
+impl std::fmt::Debug for AwsKms {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let kms_client = KmsClientDebug {
+            region: self.region.clone(),
+            endpoint: self.endpoint.clone(),
+        };
+        f.debug_struct("AwsKms")
+            .field("client", &kms_client)
+            .field("current_key_id", &self.current_key_id)
+            .field("runtime", &self.runtime)
```

Comment:
nit: just have `region`, `endpoint` and `current_key_id` as fields, no need to define a `KmsClientDebug` type and not printing runtime?

Reply:
do you have a sample output for the `runtime`?